tf2/src/buffer_core.cpp

```diff
@@ -539,20 +539,25 @@ tf2::TF2Error BufferCore::walkToTopParent(
   f.finalize(FullPath, time);
   if (frame_chain) {
     // Pruning: Compare the chains starting at the parent (end) until they differ
-    int m = static_cast<int>(reverse_frame_chain.size() - 1);
-    int n = static_cast<int>(frame_chain->size() - 1);
-    for (; m >= 0 && n >= 0; --m, --n) {
+    size_t m = reverse_frame_chain.size();
+    size_t n = frame_chain->size();
+    while (m > 0u && n > 0u) {
+      --m;
+      --n;
       if ((*frame_chain)[n] != reverse_frame_chain[m]) {
         break;
       }
     }
     // Erase all duplicate items from frame_chain
-    if (n > 0) {
-      frame_chain->erase(frame_chain->begin() + (n - 1), frame_chain->end());
+    if (n > 0u) {
+      frame_chain->erase(frame_chain->begin() + (n - 1u), frame_chain->end());
     }
 
-    if (m < reverse_frame_chain.size()) {
-      for (int i = m; i >= 0; --i) {
+    if (m < reverse_frame_chain.size())
+    {
```

Comment:
Style; I think this should be:

```suggestion
    if (m < reverse_frame_chain.size()) {
```

(we don't have the linters on in this package due to some outstanding problems; sorry about that)

Reply:
Done in https://github.com/ros2/geometry2/pull/322/commits/606b511024633a6ceca341fbe99598fdc93cf5e5.